jetty-jmx/src/main/java/org/eclipse/jetty/jmx/MBeanContainer.java

```diff
@@ -217,8 +217,11 @@ private static MetaData newMetaData(MBeanContainer container, Class<?> klass)
 
     private static Constructor<?> findConstructor(Class<?> klass)
     {
-        String pName = klass.getPackage().getName();
-        String cName = klass.getName().substring(pName.length() + 1);
+        Package pkg = klass.getPackage();
+        if (pkg == null)
+            return null;
```

Comment:
Is there no other way to return the constructor?
Or is this specifically for the constructor on the `<package>.jmx.<className>MBean` usage?

Reply:
This is the constructor specific for `*.jmx.*MBean`.